hafnian/_hermite_multidimensional.py

```diff
@@ -0,0 +1,135 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Hermite Multidimensional Python interface
+"""
+from itertools import product
+import numpy as np
+
+from .lib.libhaf import hermite_multidimensional as hm
+from ._hafnian import input_validation
+
+
+def return_prod(C, index):
+    """ Given an array C_{i,j} and an array or list of indices index = [i_1,i_2,i_3,...,i_n] it returns :math:`prod_{k=1}^n C_{1,i_1}`.
+    Args:
+        C (array): An array
+        index (array): A set of indices
+    Returns:
+        complex: The product of the array elements determines by index
+    """
+    return np.prod([C[mode, val] for mode, val in enumerate(index)])
+
+
+def expansion_coeff(alpha, resolution, renorm=True):
+    """
+    Returns the (quasi) geometric series as a vector with components alpha^i/sqrt(i!) for 0 <= i < resolution.
+    If renorm is false it omits the division by factorials
+    Args:
+        alpha (complex): Ratio of the geometric series
+        resoluton (int): Cutoff of the geometric series
+        renor (bool): Decides whether to normalize by the factorials
+    Returns:
+        array: The power series
+    """
+    vals = np.empty([resolution], dtype=type(alpha))
+    vals[0] = 1.0
+    if renorm:
+        for i in range(1, resolution):
+            vals[i] = vals[i - 1] * alpha / np.sqrt(i)
+    else:
+        for i in range(1, resolution):
+            vals[i] = vals[i - 1] * alpha
+    return vals
+
+
+def hermite_multidimensional(R, resolution, y=None, renorm=False, make_tensor=True):
+    r"""
+    Returns the multidimensional Hermite polynomials :math:`H_k^{(R)}(y)`. Here :math:`R` is an :math:n \times n: square matrix,
+    :math:`y` is an :math:`n` dimensional vector. The polynomials are parametrized by the multi-index :math:`k=(k_0,k_1,\ldots,k_{n-1})
+    and are calculated for all values :math:`0 \leq k_j < \text{resolution}`, thus a tensor of dimensions :math:`\text{resolution}^n` is returned.
+    This tensor can either be flattened into a vector or returned as an actual tensor with :math:`n` indices.
+    Note that is R = np.array([[1.0+0.0j]]) then :math:`H_k^{(R)}(y)` are precisely the well known **probabilists' Hermite polynomials** :math:`He_k(y)`:,
+    and if R = 2*np.array([[1.0+0.0j]]) then :math:`H_k^{(R)}(y)` are precisely the well known **physicists' Hermite polynomials** :math:`H_k(y)`:.
+    Args:
+        R (array): Square matrix parametrizing the Hermite polynomial family
+        resolution (int): Maximum size of the subindices in the Hermite polynomial
+        y (array): Vector for the argument of the Hermite polynomial
+        renorm (bool): If True returns :math:`H_k^{(R)}(y)/\prod(\prod_i k_i!)`
+        make_tensor: If False returns a flattened one dimensional array instead of a tensor with the values of the polynomial
+    Returns:
+        (array): The multidimensional Hermite polynomials.
```

Comment:
Note that you'll need blank lines in three places in order for this to render properly in the docs.

1. After the first line of the docstring. So after "Returns the multidimensional Hermite polynomials :math:`H_k^{(R)}(y)`.", there should be an empty line.

2. Before "Args:"

3. Before "Returns"